lib/SILGen/SILGen.cpp

```diff
@@ -1197,9 +1197,16 @@ void SILGenModule::emitOrDelayFunction(SILDeclRef constant) {
 
   auto emitAfter = lastEmittedFunction;
 
+  // When compiling at Onone, getters shouldn't be delayed, even if they don't
+  // have user written code because they can still be accessible in the
+  // debugger.
+  bool isOnoneGetter =
+      constant.isGetter() &&
+      M.getOptions().OptMode == OptimizationMode::NoOptimization;
+
   // Implicit decls may be delayed if they can't be used externally.
   auto linkage = constant.getLinkage(ForDefinition);
-  bool mayDelay = !constant.hasUserWrittenCode() &&
+  bool mayDelay = !isOnoneGetter && !constant.hasUserWrittenCode() &&
```

Comment:
Maybe hasUserWrittenCode() should also return true for property wrappers?

Reply:
I thought about that too, since @hamishknight wrote it maybe they can chime in if that'd be appropriate or not.

Reply:
The intention behind `hasUserWrittenCode` was "does this decl contain any executable code that appears in a source file?", which is needed for code coverage since we don't want to emit coverage maps for anything that won't have any source regions. It's also used here because that check is also what we need for "do we need to emit this for SILOptimizer diagnostics?". It seems like we need a different check here for the debugger which is more "can this decl be referenced from user-written code?". I wonder if we can leverage the `isSynthesized` Decl flag for that, IIRC that's effectively what it models for implicit decls.

Reply:
Out of curiosity, should this behavior also be limited to cases where we're emitting debug info?

Reply:
> Out of curiosity, should this behavior also be limited to cases where we're emitting debug info?

We usually avoid debug info interfering with code generation.